### PR TITLE
fix: Upgrade dependencies, use strict and fix /blocks pathing

### PR DESCRIPTION
**Pull request by dignifiedquire**
For @nginnever :)

### CODE REVIEW

**Comment by diasdavid** [comment]
what was the pathing issue?

File: src/stores/datastore.js [hunk 1]
```diff
@@ -1,35 +1,37 @@
+'use strict'
+
 const PREFIX_LENGTH = 8
 
 exports = module.exports
 
 exports.setUp = (basePath, blobStore, locks) => {
-  var store = blobStore(basePath + '/blocks')
+  const store = blobStore(basePath + '/blocks')
```
Comment (dignifiedquire) [inline src/stores/datastore.js]: It was here, in the version that was installed from npm, looks like I didn't fix it but the fix is not yet released to npm

Comment (dignifiedquire) [inline src/stores/datastore.js]: it said before `/blocks/`

Comment (diasdavid) [inline src/stores/datastore.js]: I'm confused

**Comment by diasdavid** [comment]
@dignifiedquire I'm still unaware of what was the bug this PR is fixing and what is the actual solution. Maybe @nginnever knows the answer?

**Comment by dignifiedquire** [comment]
Sorry will explain tomorrow

**Comment by dignifiedquire** [comment]
@diasdavid the most important fix is adding  `'use strict'` to all files, as the combination of non strict mode with some es6 features like `const` inside some browsers resulted in unpredictable behaviour when not transpiled through babel. (Remember those random tests failing for @nginnever, that was the reason for that)

The other issue was that in an older revision the repo path for the blocks was `/blocks/` but in newer revisions it is `/blocks`. But that was fixed before this PR, but wasn't published to npm.

**Comment by diasdavid** [comment]
Thank you @dignifiedquire . Also good discussion on the call today @dignifiedquire @nginnever  :)

**Pull request merged**